src/devices/src/pseudo/persist.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Defines the structures needed for saving/restoring block devices.
+
+use std::io;
+
+use snapshot::Persist;
+use utils::time::TimestampUs;
+use versionize::{VersionMap, Versionize, VersionizeResult};
+use versionize_derive::Versionize;
+use vm_memory::GuestMemoryMmap;
+
+use super::*;
+
+// No value in persisting boot timer information
+// This module restores a fresh BootTimer device.
+#[derive(Versionize)]
+pub struct BootTimerState {
+    start_ts: String,
+}
+
+pub struct BootTimerConstructorArgs {
+    pub mem: GuestMemoryMmap,
+}
+
+impl Persist<'_> for BootTimer {
+    type State = BootTimerState;
```

Comment:
Is it possible to define `State` as `()` and avoid defining a dummy `BootTimerState`?
```suggestion
    type State = ();
```
If that doesn't work, just use an empty `struct BootTimerState {}`

Same for `BootTimerConstructorArgs`.